File: test-fixtures/v151-edge-bug-repro/namespaces.py
# v151 Bug Reproduction: Python Module Paths
# Python uses . for module paths (not :: so LESS affected)
# Included for completeness - Python should work correctly

import os
import sys
from typing import Optional, List, Dict
from dataclasses import dataclass

# Direct module access patterns
import collections.abc
import urllib.parse
import json.decoder


@dataclass
class User:
    name: str
    email: str = ""


class UserService:
    def create_user(self) -> User:
        # Edge 1: Qualified module function
        path = os.path.join("/tmp", "users")

        # Edge 2: Nested module access
        parsed = urllib.parse.urlparse("http://example.com")

        # Edge 3: sys module access
        sys.stdout.write("Creating user\n")

        # Edge 4: collections.abc usage
        isinstance([], collections.abc.Sequence)

        user = User(name="test")
        self._process_user(user)
        return user

    def _process_user(self, user: User) -> None:
        # Edge 5: json module nested access
        try:
            json.decoder.JSONDecodeError
        except:
            pass

        # Edge 6: os.path nested call
        os.path.exists("/tmp")

        # Edge 7: typing module (if accessed directly)
        annotations: Dict[str, str] = {}

        print(f"Processing: {user.name}")


def main():
    service = UserService()
    user = service.create_user()

    # Edge 8: sys.exit
    sys.exit(0)


if __name__ == "__main__":
    main()

File: test-fixtures/v151-edge-bug-repro/namespaces.java
// v151 Bug Reproduction: Java Qualified Names
// Java uses . for package paths (not :: so LESS affected)
// Included for completeness - Java should work correctly

package com.myapp.services;

import java.util.ArrayList;
import java.util.List;
import java.util.HashMap;
import java.util.Map;

public class UserService {

    public User createUser() {
        // Edge 1: Fully qualified class instantiation
        java.util.ArrayList<String> list = new java.util.ArrayList<>();

        // Edge 2: Qualified static method
        String value = java.lang.System.getenv("PATH");

        // Edge 3: Nested package type
        java.util.Map<String, Object> map = new java.util.HashMap<>();

        // Edge 4: Qualified exception
        try {
            processData();
        } catch (java.lang.RuntimeException e) {
            java.lang.System.err.println(e.getMessage());
        }

        User user = new User();
        processUser(user);
        return user;
    }

    private void processUser(User user) {
        // Edge 5: Static method on qualified class
        java.lang.System.out.println("Processing user");

        // Edge 6: Qualified utility class
        java.util.Objects.requireNonNull(user);

        // Edge 7: Thread from java.lang
        java.lang.Thread.currentThread().getName();
    }

    private void processData() {
        // Simple method
    }
}

class User {
    private String name;

    public String getName() {
        return name;
    }
}
